### PR TITLE
Update signalfx exporter translations

### PR DESCRIPTION
**Pull request by dmitryax**
- Rename more k8s specific dimensions and metrics
- Use cpu.num_processors instead of machine_cpu_cores
- Compute memory.total
- Split system.memory.usage

### CODE REVIEW

File: exporter/signalfxexporter/translation/constants.go [hunk 1]
```diff
@@ -48,25 +53,44 @@ translation_rules:
...
+    k8s/job/active_pods: kubernetes.job.active
+    k8s/job/desired_successful_pods: kubernetes.job.completions
+    k8s/job/failed_pods: kubernetes.job.failed
+    k8s/job/max_parallel_pods: kubernetes.job.parallelism
+    k8s/job/successful_pods: kubernetes.job.succeeded
+    k8s/namespace/phase: kubernetes.namespace_phase
     k8s/node/condition_memory_pressure: kubernetes.node_memory_pressure
     k8s/node/condition_network_unavailable: kubernetes.node_network_unavailable
+    k8s/node/condition_out_of_disk: kubernetes.node_out_of_disk
+    k8s/node/condition_p_i_d_pressure: kubernetes.node_p_i_d_pressure
```
Comment (jrcamp) [inline exporter/signalfxexporter/translation/constants.go:85]: Weird, is `k8s/node/condition_p_i_d_pressure` getting emitted? Should probably be `k8s/node/condition_pid_pressure`  @asuresh4

File: exporter/signalfxexporter/translation/constants.go [hunk 2]
```diff
@@ -119,14 +143,43 @@ translation_rules:
 
 - action: copy_metrics
   mapping:
-    cpu.idle: machine_cpu_cores
```
Comment (jrcamp) [inline exporter/signalfxexporter/translation/constants.go:122]: Why is cpu.idle getting mapped to number cores? Isn't it converted in previous rule?

File: exporter/signalfxexporter/translation/constants.go (hunk 1 above)
Comment (dmitryax) [reply exporter/signalfxexporter/translation/constants.go:85]: I'm going to handle this as the next PR with updating k8s cluster receiver to otel convention

File: exporter/signalfxexporter/translation/constants.go (hunk 2 above)
Comment (dmitryax) [reply exporter/signalfxexporter/translation/constants.go:122]: it's a step to prepare for computation of `cpu.num_processors` in the next rule

Comment (dmitryax) [reply exporter/signalfxexporter/translation/constants.go:122]: let me put a comment here

**Review by jrcamp — APPROVED** [review]

File: exporter/signalfxexporter/translation/constants.go (hunk 1 above)
Comment (asuresh4) [reply exporter/signalfxexporter/translation/constants.go:85]: This is likely an artifact of the helper that converts camelcase to snakecase.

**Pull request merged**